aspnetcore/performance/caching/middleware/samples/6.x/ResponseCachingMiddleware/Program.cs

```diff
@@ -41,10 +31,45 @@
     await next();
 });
 
-app.UseEndpoints(endpoints =>
+app.MapGet("/", () => DateTime.Now.Millisecond);
+
+app.Run();
+#endregion
+#elif SECOND
+#region snippet4
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddResponseCaching(options =>
 {
-    endpoints.MapRazorPages();
+    options.MaximumBodySize = 1024;
+    options.UseCaseSensitivePaths = true;
 });
 
+var app = builder.Build();
+
+app.UseHttpsRedirection();
+
+// UseCors must be called before UseResponseCaching
+//app.UseCors();
+
+app.UseResponseCaching();
+
+app.Use(async (context, next) =>
+{
+    context.Response.GetTypedHeaders().CacheControl =
+        new Microsoft.Net.Http.Headers.CacheControlHeaderValue()
+        {
+            Public = true,
+            MaxAge = TimeSpan.FromSeconds(10)
+        };
+    context.Response.Headers[Microsoft.Net.Http.Headers.HeaderNames.Vary] =
+        new string[] { "Accept-Encoding" };
+
+    await next();
+});
```

Comment:
This can use the [new overload](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-6.0#prefer-appuse-overload-that-requires-passing-the-context-to-next):

```suggestion
app.Use(async (context, next) =>
{
    context.Response.GetTypedHeaders().CacheControl =
        new Microsoft.Net.Http.Headers.CacheControlHeaderValue()
        {
            Public = true,
            MaxAge = TimeSpan.FromSeconds(10)
        };
    context.Response.Headers[Microsoft.Net.Http.Headers.HeaderNames.Vary] =
        new string[] { "Accept-Encoding" };

    await next(context);
});
```